test/JDBC/expected/remote-proc-exec-vu-verify.out

```diff
@@ -0,0 +1,1211 @@
+GO
+~~START~~
+int#!#bigint#!#smallint#!#tinyint
+2147483647#!#9223372036854775807#!#32767#!#255
+~~END~~
+
+
+-- Test 74: Precision boundary for DECIMAL
+-- Expected: Should handle maximum precision
+DECLARE @max_decimal DECIMAL(18,4) = 99999999999999.9999;
+EXEC bbf_rpe_server.master.dbo.sp_TestDecimal @p_decimal=@max_decimal, @p_numeric=99999999.99;
+GO
+~~START~~
+numeric#!#numeric
+99999999999999.9999#!#99999999.99
+~~END~~
+
+
+-- Test 75: String with only spaces
+-- Expected: Should preserve spaces
+EXEC bbf_rpe_server.master.dbo.sp_TestVarchar @p='     ';
+GO
+~~START~~
+varchar#!#int
+     #!#0
+~~END~~
+
+
+-- Test 76: String with quotes inside
+-- Expected: Should handle escaped quotes
+EXEC bbf_rpe_server.master.dbo.sp_TestVarchar @p='It''s a test';
+GO
+~~START~~
+varchar#!#int
+It's a test#!#11
+~~END~~
+
+
+-- Test 77: NVARCHAR with Unicode characters (if supported)
+-- Expected: Should preserve Unicode
+DECLARE @unicode NVARCHAR(100) = N'Hello 世界 مرحبا';
+EXEC bbf_rpe_server.master.dbo.sp_TestNvarchar @p=@unicode;
+GO
+~~START~~
+varchar#!#int
+Hello ?? ?????#!#14
+~~END~~
+
+
+
+-- ========================================
+-- SECTION 14: COMPLEX SCENARIOS
+-- ========================================
+-- Test 78: OUTPUT with conditional logic
+-- Expected: Should set different output based on input
+DECLARE @cond_out INT;
+EXEC bbf_rpe_server.master.dbo.sp_OutputSingle @input=100, @output=@cond_out OUTPUT;
+SELECT @cond_out AS conditional_output;
+GO
+~~START~~
+int
+200
+~~END~~
+
+~~START~~
+int
+200
+~~END~~
+
+
+-- Test 79: Multiple sequential remote calls
+-- Expected: Each should execute independently
+EXEC bbf_rpe_server.master.dbo.sp_SingleInt @value=1;
+EXEC bbf_rpe_server.master.dbo.sp_SingleInt @value=2;
+EXEC bbf_rpe_server.master.dbo.sp_SingleInt @value=3;
+GO
+~~START~~
+int#!#int
+1#!#2
+~~END~~
+
+~~START~~
+int#!#int
+2#!#4
+~~END~~
+
+~~START~~
+int#!#int
+3#!#6
+~~END~~
+
+
+-- Test 80: OPENQUERY with stored procedure (documents unsupported behavior)
+-- Note: OPENQUERY with EXEC does not return result sets - this is expected to fail
+-- This test documents the limitation, not a bug
+DECLARE @val INT;
+SELECT @val = (SELECT TOP 1 number FROM OPENQUERY(bbf_rpe_server, 'EXEC sp_NoParams'));
+SELECT @val AS assigned_value;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Query passed to OPENQUERY did not return a result set)~~
+
+
+
+-- ========================================
+-- SECTION 15: SECURITY VALIDATION
+-- ========================================
+-- Test 81: Verify SELECT-only procedure works
+-- Expected: Should execute successfully
+EXEC bbf_rpe_server.master.dbo.sp_OnlySelect;
+GO
+~~START~~
+varchar#!#int
+This should work#!#1
+~~END~~
+
+
+-- Test 82: SQL Injection attempt in procedure name (should fail safely)
+-- Expected: Should throw error - procedure not found (not SQL injection)
+EXEC bbf_rpe_server.master.dbo.[sp_NoParams'; DROP TABLE test; --];
```

Comment:
Please try injection in server name, database name and procedure name. Also, I don't think the procedure name injection is actually getting tested here. Having `'` in a procedure name is perfectly valid in T-SQL if it wrapped in `[]`